[4805] Create image editor control

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4805

aachavarria committed:
adding dataSources to ContentType;
adding imageDatasources validations;

aachavarria committed:
adding getContentTypeOptions for EditFormPanel.tsx

aachavarria committed:
Creating BrowseFilesDialog.tsx

aachavarria committed:
adding data sources to asset drop

aachavarria committed:
refactor on DESKTOP_ASSET_DROP

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialog.tsx:
```diff
@@ -0,0 +1,47 @@
+import { BrowseFilesDialogUI } from './BrowseFilesDialogUI';
+import { FormattedMessage } from 'react-intl';
+
+export interface BrowseFilesDialogProps {
```
Extend dialog props and use them

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogUI.tsx:
```diff
@@ -0,0 +1,76 @@
+      console.log(response.items);
+      setItems(response.items);
+    });
+  }, [path, site]);
```
We should look at how the legacy browse dialog works and what it does.

aachavarria replied:
it is a new approach, and I think it is better because is using api2

rart replied:
Not suggesting to change it just yet. Just check what the old one does to make sure this covers all of that.

rart commented on ui/app/src/components/MediaCard.tsx:
```diff
@@ -247,7 +246,7 @@ function MediaCard(props: MediaCardProps) {
           avatar={Avatar ? <Avatar /> : null}
           classes={{ root: classes.cardHeaderRoot, avatar: classes.avatar }}
-          onClick={() => onPreview(item)}
+          onClick={() => onPreview && onPreview(item)}
```
```suggestion
          onClick={() => onPreview?.(item)}
```

aachavarria committed:
refactor on DESKTOP_ASSET_DROP

rart commented on ui/app/src/utils/path.ts:
```diff
@@ -205,3 +205,26 @@ export function getCodeEditorSrc({
+    let year = String(currentDate.getFullYear());
+    path = path.replace('{year}', year);
+    path = path.replace('{yyyy}', year);
+  }
```
```suggestion
path = path.replace(/{(year|yyyy)}/g, String(currentDate.getFullYear()));
```
same deal for the ones below. No need for if.

rart commented on ui/app/src/utils/path.ts:
```diff
@@ -205,3 +205,26 @@ export function getCodeEditorSrc({
   return `${authoringBase}/legacy/form${qs}`;
 }
+
+export function expandPathMacros(path: string) {
```
Let's go with the `values` argument that would internally call another `format` util.

aachavarria committed:
adding utils to craftercms.ts

aachavarria committed:
fixing issue on img-CMIS-upload.js when parsing values from fileData;
adding uploader support to image DnD

aachavarria committed:
Creating SingleFileUploadDialog

rart commented on ui/app/src/utils/path.ts:
```diff
@@ -205,3 +206,21 @@ export function getCodeEditorSrc({
+  path = path
+    .replace(/{(year|yyyy)}/g, String(currentDate.getFullYear()))
+    .replace(/{(month|mm)}/g, ('0' + (currentDate.getMonth() + 1)).slice(-2))
+    .replace(/{(dd)}/g, ('0' + currentDate.getDate()).slice(-2));
```
```suggestion
    .replace(/{dd}/g, ('0' + currentDate.getDate()).slice(-2));
```

rart commented on ui/app/src/services/content.ts:
```diff
@@ -592,7 +592,6 @@ function insertCollectionItem(
 export function createFileUpload(
   uploadUrl: string,
   file: any,
-  path: string,
```
How come this isn't used? How does it know where to upload?

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialog.tsx:
```diff
@@ -0,0 +1,49 @@
+}
+
+export default function BrowseFilesDialog(props: BrowseFilesDialogProps) {
+  const { open, path, type, onClose, onClosed, ...rest } = props;
```
```suggestion
  const { path, type, onClosed, ...rest } = props;
```

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialog.tsx:
```diff
@@ -0,0 +1,49 @@
+  const { open, path, type, onClose, onClosed, ...rest } = props;
+
+  return (
+    <Dialog open={open} onClose={onClose} {...rest} fullWidth maxWidth="md">
```
```suggestion
    <Dialog fullWidth maxWidth="md" {...rest}>
```

aachavarria committed:
Fixing BrowseFilesDialog props

aachavarria committed:
Adding singleFileUpload.ts to GlobalDialogManager.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4805

 Conflicts:
	ui/app/src/modules/Preview/Tools/EditFormPanel.tsx

aachavarria committed:
Adding logic to upload files using SingleFileUploadDialog.tsx

aachavarria committed:
Refactors on SingleFileUploadDialog to add validTypes, profile id and uploadType
Integrating Cmis,S3, webDav to SingleFileUploadDialog.tsx
Creating notifyFieldImageChanged action to notify from host to guest

aachavarria committed:
Adding BrowseFilesDialog.tsx to GlobalDialogManager.tsx
Creating actions/reducer for BrowseFilesDialog.tsx

aachavarria committed:
adding contentTypes, mimeTypes, multiSelect;
adding skeleton;

aachavarria committed:
adding multiSelect on success

aachavarria committed:
adding delete support

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4805

 Conflicts:
	ui/app/src/components/SystemStatus/GlobalDialogManager.tsx
	ui/app/src/models/GlobalState.ts
	ui/app/src/state/actions/dialogs.ts

rart commented:
Will revisit this when we're ready to proceed. Please keep the branch.
